src/genworld_gui.cpp

```diff
@@ -375,6 +389,38 @@ static DropDownList BuildTownNameDropDown()
 	return list;
 }
 
+/**
+ * Set string parameters before drawing the number of minutes per year.
+ * @param max Set the maximum value, to determine widget size.
+ */
+static void SetMinutesPerYearParams(bool max = false)
+{
+	/* When resizing widgets, the longest string is 1% of default. */
+	int minutes = (max ? (CalendarTime::DEF_MINUTES_PER_YEAR * 100) : _settings_newgame.economy.minutes_per_calendar_year);
+
+	if (minutes == CalendarTime::DEF_MINUTES_PER_YEAR) {
+		/* Calendar time is default. */
+		SetDParam(0, STR_CONFIG_SETTINGS_MINUTES_PER_YEAR_VALUE_DEFAULT);
+		SetDParam(1, minutes);
+	} else if (minutes == CalendarTime::FROZEN_MINUTES_PER_YEAR) {
+		/* Calendar time is frozen. */
+		SetDParam(0, STR_CONFIG_SETTINGS_MINUTES_PER_YEAR_VALUE_FROZEN);
+	} else {
+		/* Calendar time is not default or frozen. Let's try to show a percent of default game speed. */
+		int percent = 100 * CalendarTime::DEF_MINUTES_PER_YEAR / minutes;
+
+		if (percent > 0) {
+			/* Only draw a percent if it's above 0%. */
+			SetDParam(0, STR_CONFIG_SETTINGS_MINUTES_PER_YEAR_VALUE_PERCENT);
+			SetDParam(1, minutes);
+			SetDParam(2, percent);
+		} else {
+			SetDParam(0, STR_CONFIG_SETTING_MINUTES_PER_YEAR_VALUE);
+			SetDParam(1, minutes);
+		}
+	}
```

Comment:
This if-cascade is rather hard to read, and hard to figure out what it actually does. As in: I get what the code does, but why it is this complicated I do not really understand? Some suggestions:

Start with the exception and early return:

`if (minutes == frozen) { SetDParam(..); return; }`

That way, we got that one out of the way.

Next, there is something complicated about sometimes returning a percentage, sometimes a value. Also there is `SETTING` vs `SETTINGS`, where the latter has a `{BLACK}`, but the first doesn't. But both are used here.

Also, I am not sure why it is important to add the `(default)` here? I don't think we do it anywhere else?

I would reason: the default is 100%, right? So you can just always show that: `12 minutes (100%)`. I think that is enough reinforcement that that is the default?

Lastly, although I get that when percentage goes < 0.5%, you no longer want to draw %, but that feels weird too. Maybe simpler rules:

`if percentage >= 1`: `N minutes (M%)`
`if percentage == 0`: `N minutes (<1%)`

That would simplify this function a lot, and makes things a bit easier for the end-user too?